src/pages/bookReview.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState, useEffect, useCallback } from "react";
+import { useLocation } from "react-router-dom";
+import styled from "styled-components";
+import { getBookInfo } from "../api/books";
+import StarRating from "../component/starRating";
+
+const Container = styled.div`
+  margin-left: 6%;
+  margin-right: 6%;
+`;
+
+function BookReview(props) {
+  let location = useLocation();
+  const id = location.state.id;
+
+  const [bookInfo, setBookInfo] = useState([]);
+
+  const getBookById = useCallback(async (id) => {
```

Comment:
I can that this code is repeated multiple times in multiple files. This would be a great candidate for a custom hook. Please create a `hooks` folder and create this custom hook to get a book by it's id